prow/version/metrics.go

```diff
@@ -30,6 +32,10 @@ var (
 
 func init() {
 	prometheus.MustRegister(prowVersion)
+	// The line above already emits a prow_version metrics with 0 value, which
+	// is inaccurate. Since the version would not change for the running binary,
+	//  doing this once when binary starts should fix this problem.
+	GatherProwVersion(logrus.WithContext(context.Background()))
```

Comment:
Wouldn't we need to do *before` the `MustRegister` then? :confused:

Reply:
I think `MustRegister` is required before hand, it's for registering the metrics to be collected by prometheus https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#MustRegister. I think I've done a bad job commenting this section, it was based on my observation of this metrics value being 0 in components that didn't call `GatherProwVersion`, it could just be a default value when this prometheus metric has not been reporter, let me try to rephrase

Reply:
updated